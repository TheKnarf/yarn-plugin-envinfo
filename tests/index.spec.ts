const plugin = require('../src/index.ts').default;
const { promisify } = require('util');
const exec = promisify(require('child_process').exec)

jest.setTimeout(10000);

test('Exports 1 command', () => {
	expect(plugin.commands.length).toBe(1);
});

test('Snapshot test of `yarn pack --dry-run --json`', async () => {
	const result = await exec('yarn pack --dry-run --json');

	const stderr = result.stderr;
	expect(stderr).toMatchSnapshot();

	// Skip the first line of the output since thats just the build path
	const stdout = result
		.stdout
		.split("\n")
		.slice(1)
		.join("\n");

	expect(stdout).toMatchSnapshot();
});


