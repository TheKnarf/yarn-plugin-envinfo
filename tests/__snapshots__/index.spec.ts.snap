// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test of \`yarn pack --dry-run --json\` 1`] = `""`;

exports[`Snapshot test of \`yarn pack --dry-run --json\` 2`] = `
"{\\"location\\":\\"LICENSE\\"}
{\\"location\\":\\"Readme.md\\"}
{\\"location\\":\\"index.js\\"}
{\\"location\\":\\"package.json\\"}
"
`;
